packages/router/router.ts

```diff
@@ -1108,14 +1123,15 @@ export function createRouter(init: RouterInit): Router {
     saveScrollPosition(state.location, state.matches);
     pendingPreventScrollReset = (opts && opts.preventScrollReset) === true;
 
+    let routesToUse = inFlightDataRoutes || dataRoutes;
```

Comment:
This feels sufficient for now since it's only needed in a few spots.  But I am wondering if this feels like a hard-to-detect footgun down the road if we add some new code and just reach for `dataRoutes`.  One thought was a quick getter function like `getCurrentDataRoutes()` but maybe even something like renaming `dataRoutes -> comittedRoutes` would help add some explicit semantics too.  I think it's probably best to table the decision until we tackle the follow ups for fog of war and MFE though

Reply:
Yea let's tackle this for sure when we got to the granular route updates API